homeassistant/components/light/__init__.py

```diff
@@ -228,6 +231,11 @@ def handle_light_service(service):
             params.setdefault(ATTR_XY_COLOR, profile[:2])
             params.setdefault(ATTR_BRIGHTNESS, profile[2])
 
+        color_name = params.pop(ATTR_COLOR_NAME, None)
+
+        if color_name is not None:
+            params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
```

Comment:
Can we convert this to a list before setting rgb color, or decide to change rgb color to always be a tuple. I think mixing types like this is unpredictable for devs.